### PR TITLE
Add the query string generator in back end test files

### LINKED ISSUE #216: Use "query_string_generator" in all of the back end test files

### PR DESCRIPTION
**Pull request by DarrenZhang01**
fixes #216

### CODE REVIEW

File: test/back_end/api_post_test.spec.js [hunk 1]
```diff
@@ -192,8 +282,15 @@ describe("Test POST/UPDATE/DELETE requests to API", function() {
             );
             expect(survey_instance_id1).to.satisfy(Number.isInteger);
 
+            parameter_list = {
+                what: "launch_survey",
+                user_id: "woods13",
+                survey_id: 5,
+                action: "add_or_update"
+            };
+            query_string = generate_query_string(parameter_list);
```
Comment (siefkenj) [inline test/back_end/api_post_test.spec.js]: no `let`?

File: test/back_end/api_get_test.spec.js [hunk 2]
```diff
@@ -3,12 +3,18 @@ const chai = require("chai");
...
+import generate_query_string from "../../src/views/components/generate_query_string.js";
 
 describe("Test GET requests to API", function() {
     it("should return user info for Joshua Cook", async function() {
         try {
+            let parameter_list = {
+                what: "user_info",
+                user_id: "cook19"
+            };
+            let query_string = generate_query_string(parameter_list);
```
Comment (siefkenj) [inline test/back_end/api_get_test.spec.js]: All of these can be made more compact.
```
let query_string = generate_query_string({
                what: "user_info",
                user_id: "cook19"
            });
```

File: test/back_end/api_post_test.spec.js [hunk 3]
```diff
@@ -40,9 +48,16 @@ describe("Test POST/UPDATE/DELETE requests to API", function() {
                 default_survey_close: null
             });
             let post_body = base64.encode(body);
+            let query_string = generate_query_string({
+                what: "surveys",
+                user_id: "woods13",
+                level: "dept",
+                survey_id: 1,
+                action: "add_or_update",
+                post_body: "base64:" + post_body
```
Comment (siefkenj) [inline test/back_end/api_post_test.spec.js]: Again, consolidate. `"base64:" + base64.encode(body)`

File: test/back_end/api_get_test.spec.js [hunk 4]
```diff
@@ -210,8 +264,12 @@ describe("Test GET requests to API", function() {
     });
     it("should return list of tas under Elizabeth Miller", async function() {
         try {
```
Comment (siefkenj) [inline test/back_end/api_get_test.spec.js]: Again...it is silly to have these in `try` `catch` blocks...

Comment (DarrenZhang01) [reply test/back_end/api_get_test.spec.js]: Sorry. It was not written by me. But I will revise that.

**Pull request merged**